packages/design-system/src/components/OcBreadcrumb/OcBreadcrumb.vue

```diff
@@ -142,6 +181,69 @@ export default defineComponent({
   },
   setup(props) {
     const { $gettext } = useGettext()
+    const visibleItems = ref<BreadcrumbItem[]>([])
+    const hiddenItems = ref<BreadcrumbItem[]>([])
+    const displayItems = ref<BreadcrumbItem[]>(props.items.slice())
+
+    const getBreadcrumbElement = (id): HTMLElement => {
+      return document.querySelector(`[data-item-id="${id}"]`)
+    }
+
+    const calculateTotalBreadcrumbWidth = () => {
+      let totalBreadcrumbWidth = 100 // 100px margin to the right to avoid breadcrumb from getting too close to the controls
+      visibleItems.value.forEach((item, index) => {
+        const breadcrumbElement = getBreadcrumbElement(item.id)
+        const itemClientWidth = breadcrumbElement?.getBoundingClientRect()?.width || 0
+        totalBreadcrumbWidth += itemClientWidth
+      })
+      return totalBreadcrumbWidth
+    }
+
+    const reduceBreadcrumb = (offsetIndex) => {
+      const breadcrumbMaxWidth = props.maxWidth
+      if (!breadcrumbMaxWidth) {
+        return
+      }
+      document.getElementById(props.id)?.style.setProperty('--max-width', `${breadcrumbMaxWidth}px`)
+      const totalBreadcrumbWidth = calculateTotalBreadcrumbWidth()
+
+      const isOverflowing = breadcrumbMaxWidth < totalBreadcrumbWidth
+      if (!isOverflowing || visibleItems.value.length <= 3) {
+        return
+      }
+      // Remove from the left side
+      const removed = visibleItems.value.splice(offsetIndex, 1)
+
+      hiddenItems.value.push(removed[0])
+      reduceBreadcrumb(offsetIndex)
+    }
+
+    const lastHiddenItem = computed(() =>
+      hiddenItems.value.length >= 1 ? unref(hiddenItems)[unref(hiddenItems).length - 1] : { to: {} }
+    )
+
+    const renderBreadcrumb = () => {
+      displayItems.value = props.items.slice()
+      if (displayItems.value.length > 1) {
+        displayItems.value.splice(1, 0, {
+          text: '...',
+          allowContextActions: false,
+          to: {},
+          isTruncated: true
```

Comment:
Could you please add `isAriaHidden: true` to this element and then use that to make the whole `<li>` tag `:aria-hidden="true"` in the template? This is important so that screen reader users don't get confused by the `...` list item. Their usage of the breadcrumbs is kind of "superior" to the seeing eye, because they can always read the full list of breadcrumb items (see comment about `oc-invisible-sr` css class above).

Reply:
And another one: Did you plan to use `isTruncated` in a different way? Because I only ever see it set and used for the `...` breadcrumb item. What I read from your discussion with @JanAckermann was that it was renamed to this because it was supposed to represent any of the truncated items. So `isAriaHidden: true` from my previous comment is probably not needed, you can instead rename `isTruncated` again to `isTruncationPlaceholder` or something like that. You can then use it for both checking its visibility and for setting it to `aria-hidden="true"`.